### DISCUSSION TITLE
Autosave cursor position

### DISCUSSION THREAD

**Discussion by spoeken** [react-hook-form/react-hook-form#4166] [Q&A]
Hi, I've written this autosave component and have a tiny problem.
I have set it to submit the form if no changes was done the last 1000ms, the problem occurs if the user edits in the middle of a text input. Whenever the form is reset (after submission) the cursor is set to the end of the text.

I've created a sandbox here: https://codesandbox.io/s/react-hook-form-auto-save-xgulp?file=/src/index.js

**Comment by bluebill1049** [comment]
`reset` action does not focus safely.

if you are on a new project, I would suggest using the latest v7 (which is in beta), it has a watch callback for you to subscribe to the changes, so you can probably avoid the reset action.

take a look at the watch section: https://github.com/react-hook-form/react-hook-form/discussions

**Reply by bluebill1049** [reply]
or maybe even consider using useWatch at the current version to avoid reset

**Reply by spoeken** [reply]
Hi, thank your for your answer. As you can see in the example I provided I already use the useWatch hook.

**Comment by spoeken** [comment]
For others facing this, I decided that the submission shouldn't invalidate the query for data, hence reset(defaultValues) won't happen. That way the client is the source of truth. This would be an issue if two people where editing at the same time, but in my case that won't happen
